src/solvers/elasticity_solver.cpp

```diff
@@ -13,21 +13,12 @@ namespace serac {
 constexpr int NUM_FIELDS = 1;
 
 ElasticitySolver::ElasticitySolver(int order, std::shared_ptr<mfem::ParMesh> pmesh)
-    : BaseSolver(pmesh->GetComm(), NUM_FIELDS, order), displacement_(state_[0])
+    : BaseSolver(pmesh->GetComm(), NUM_FIELDS, order),
+      displacement_(std::make_shared<FiniteElementState>(
+          order, pmesh, FESOptions{.ordering = mfem::Ordering::byVDIM, .name = "displacement"}))
```

Comment:
designated initializers aren't officially supported until C++20, right?

Reply:
well, it looks like clang and gcc support it with C++17, but MSVC doesn't: https://godbolt.org/z/MYca48

Reply:
Yes, it's not official until C++20, but most compilers (even Intel) have supported it for almost a decade.  Designated initializers were part of C99 so the porting effort for the language implementers was minimal.  Does anyone feel strongly about adding a dependency on technically-not-standardized language features?

Reply:
can't we just write a 2 argument constructor for `FESOptions` instead? By the same token, we still don't use the widely (but unofficially) supported `#pragma once` either, presumably for the same reason.

Reply:
I vote this is a pretty useful and innocuous compiler feature, that if our compiler set support it we can move ahead with it.  We can take these compiler features on a case by case basis.

Reply:
The issue is that there are (currently) four options (all with defaults) and different solvers need to specify different options, so you'd need a constructor for each item in the powerset of options (2^n = 16 ctors).  Designated initializers let you specify only what you care about and thus avoid the problem of having to worry about things like having default arguments in the right order.

Reply:
It's also easy to remove if it causes a problem in nvcc.

Reply:
Once we get a "dumb" GPU build and CI up and running, I will feel much more comfortable with these unofficial language features. The `#pragma once` would also be really nice...